controllers/remote/cluster_cache.go

```diff
@@ -225,6 +240,42 @@ func (t *ClusterCacheTracker) newClusterAccessor(ctx context.Context, cluster cl
 	}, nil
 }
 
+func runningOnCluster(ctx context.Context, c client.Client, cluster client.ObjectKey) (bool, error) {
+	podNamespace := os.Getenv("POD_NAMESPACE")
+	podName := os.Getenv("POD_NAME")
+	podUID := os.Getenv("POD_UID")
+	if podNamespace == "" || podName == "" || podUID == "" {
+		return false, nil
```

Comment:
Log here that we couldn't access any of these variables?

Reply:
I moved this section so we do it only once per ClusterCacheTracker and log it there.

Otherwise we would log this for every workload cluster (and we only actually have to read the env vars once)